### DISCUSSION TITLE
Is this library able to concurrently send email?

### DISCUSSION THREAD

**Discussion by Dennis-Zhang-SH** [wneessen/go-mail#135] [Q&A] [answered]
I want to send email concurrently.

**Comment by Dennis-Zhang-SH** [comment]
I just tested it, it seems no.

**Reply by wneessen** [reply, marked as answer]
Well, Email per definition does not exactly provide a concurrency model itself. Mail server accept mail commands sequencially, add the provided mail to their internal queue and then deliver them.

Our `Client` allows for providing multiple `Msg` at once for the `DialAndSend` method, which then will provided sequencially, but in one call, to the mail server. Alternatively you could set up go routines that create a new `Client` and then run those go routines concurrently. But go-mail itself does, at least currently, not provide the appropriate tooling for "cuncurrency".

**Reply by Dennis-Zhang-SH** [reply]
okay
